tests/System/MultiRoute/DonationAndCancellationTest.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Tests\System\MultiRoute;
+
+use Symfony\Component\HttpKernel\Client;
+use WMDE\Fundraising\Frontend\Factories\FunFunFactory;
+use WMDE\Fundraising\Frontend\Tests\System\Routes\AddDonationRouteTest;
+use WMDE\Fundraising\Frontend\Tests\System\WebRouteTestCase;
+
+/**
+ * @licence GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+class DonationAndCancellationTest extends WebRouteTestCase {
+
+	const UPDATE_TOKEN_COOKIE_NAME = 'wmde-fundraising-utoken';
+	const DONATION_ID_COOKIE_NAME = 'wmde-fundraising-donation-id';
+
+	public function testWhenCreatingDonation_itCanBeCancelled() {
+		$this->createEnvironment( [], function ( Client $client, FunFunFactory $factory ) {
+			$factory->setNullMessenger();
+
+			$client->request(
+				'POST',
+				'/donation/add',
+				AddDonationRouteTest::newValidFormInput()
+			);
+
+			$this->assertSame( 200, $client->getResponse()->getStatusCode() );
+
+			$client->request(
+				'POST',
+				'/donation/cancel',
+				[
+					'sid' => $client->getCookieJar()->get( self::DONATION_ID_COOKIE_NAME )->getValue(),
```

Comment:
I actually was thinking that the cancel route as we implemented it here so far as using URL parameters, and was planning to ask you or Kai about that, as it seemed rather odd when I was working on this cookie stuff. So I'm winning the "most confoosed" badge :)

Looking at what is going on in `Request`, the `get` method which we're using in the routes will use GET and POST params, with the former getting precedence. No cookies though. (Not 100% sure I understood the code correctly.)

So we probably want to change that, and also stop using `get`, in favour of more explicit accessors. (Which is actually recommended by the `get` docs). Let's do that in a different PR tho.